src/ansys/fluent/core/utils/deprecate_args.py

```diff
@@ -0,0 +1,33 @@
+"""Module that provides a method to handle deprecated arguments."""
+
+import functools
+import warnings
+
+
+def deprecate_argument(
+    old_arg, new_arg, converter, deprecation_class=DeprecationWarning
+):
+    """Warns user that the argument provided is deprecated, and automatically replaces
+    the deprecated argument with the appropriate new argument."""
+
+    def decorator(func):
+        """Holds the original method to perform operations on it."""
+
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            """Warns about the deprecated argument and replaces it with the new
+            argument."""
+            if old_arg in kwargs:
+                warnings.warn(
+                    f"'{old_arg}' is deprecated. Use '{new_arg}' instead.",
+                    deprecation_class,
+                )
+                val = converter(kwargs[old_arg])
```

Comment:
A slightly more verbose variable name would improve code clarity here I think, something like:
```suggestion
                new_value = converter(kwargs[old_arg])
```